core/adapters/base.ts

```diff
@@ -22,18 +23,50 @@ export abstract class Adapter {
 
   protected insertInto(target: dataform.ITarget, columns: string[], query: string) {
     return `
-      insert into ${this.resolveTarget(target)}
-      (${columns.join(",")})
-      select ${columns.join(",")}
-      from (${query}) as insertions`;
+insert into ${this.resolveTarget(target)}
+(${columns.join(",")})
+select ${columns.join(",")}
+from (${query}) as insertions`;
   }
 
   protected oppositeTableType(type: string) {
     return this.baseTableType(type) === "table" ? "view" : "table";
   }
 
   protected where(query: string, where: string) {
-    return `select * from (${query}) as subquery
-        where ${where || "true"}`;
+    return `
+  select * from (${query}) as subquery
+    where ${where || "true"}`;
+  }
+
+  protected refreshIncrementCondition(
```

Comment:
i would suggest inverting this to `!runConfig.fullRefresh && tableMetadata && tableMetadata.type !== "view"`, and call it `shouldWriteIncrementally`

Reply:
Sure